core/src/autogluon/core/utils/utils.py

```diff
@@ -360,12 +360,115 @@ def get_pred_from_proba(y_pred_proba, problem_type=BINARY):
     return y_pred
 
 
+def extract_label(data: DataFrame, label: str) -> (DataFrame, Series):
+    """
+    Extract the label column from a dataset and return X, y.
+
+    Parameters
+    ----------
+    data : DataFrame
+        The data containing features and the label column.
+    label : str
+        The label column name.
+
+    Returns
+    -------
+    X, y : (DataFrame, Series)
+        X is the data with the label column dropped.
+        y is the label column as a pd.Series.
+    """
+    if label not in list(data.columns):
+        raise ValueError(f"Provided DataFrame does not contain label column: {label}")
+    y = data[label].copy()
+    X = data.drop(label, axis=1)
+    return X, y
+
+
+def generate_train_test_split_combined(data: DataFrame,
+                                       label: str,
+                                       problem_type: str,
+                                       test_size: float = 0.1,
+                                       random_state: int = 0,
+                                       min_cls_count_train: int = 1) -> (DataFrame, DataFrame):
+    """
+    Generate a train test split from a DataFrame that contains the label column.
+
+    Parameters
+    ----------
+    data : DataFrame
+        DataFrame containing the features plus the label column to split into train and test sets.
+    label : str
+        The label column name.
+        Used for stratification and to ensure all classes in multiclass classification are preserved in train data.
+    problem_type : str
+        The problem_type the label is used for. Determines if stratification is used.
+        Options: ["binary", "multiclass", "regression", "softclass", "quantile"]
```

Comment:
nit: we don't enforce this list anywhere. Add defensive assert here to ensure that we guard against typos?

Reply:
will add to follow-up PR